core/thread/inc/ThreadPool.h

```diff
@@ -0,0 +1,106 @@
+#ifndef THREADPOOL_H_
```

Comment:
this file is missing the normal header and is not following the code guard convention.